community/cypher/cypher-logical-plans-3.4/src/main/scala/org/neo4j/cypher/internal/v3_4/logical/plans/AssertSameNode.scala

```diff
@@ -21,11 +21,21 @@ package org.neo4j.cypher.internal.v3_4.logical.plans
 
 import org.neo4j.cypher.internal.ir.v3_4.{CardinalityEstimation, IdName, PlannerQuery}
 
+/**
+  * For every row in left, assert that all rows in right produce the same value
+  * for the variable IdName. Produce the rows from left.
+  *
+  * for ( leftRow <- left )
+  *   for ( rightRow <- right )
+  *     assert( leftRow(node) == rightRow(node) )
+  *   produce leftRow
+  */
+// TODO: rename this to MergeSomething...
```

Comment:
Hmm, I was just wanting to associate it closer to the rest of the merge operators, because that's the only case where it's used? 

Added a line about that instead. Just thought the operator was curious, so wanted to explain somehow why it exists.